joy_teleop/scripts/joy_teleop.py

```diff
@@ -188,14 +188,24 @@ def run_topic(self, c, joy_state):
 
         else:
             for mapping in cmd['axis_mappings']:
-                if len(joy_state.axes)<=mapping['axis']:
-                  rospy.logerr('Joystick has only {} axes (indexed from 0), but #{} was referenced in config.'.format(len(joy_state.axes), mapping['axis']))
-                  val = 0.0
+                if 'axis' in mapping:
+                    if len(joy_state.axes) <= mapping['axis']:
+                        rospy.logerr('Joystick has only {} axes (indexed from 0), but #{} was referenced in config.'.format(len(joy_state.axes), mapping['axis']))
+                        val = 0.0
+                    else:
```

Comment:
could you flip the logic here please? 

it is more intuitive to do it with the cases you intend to cover first, then error out on the rest